### PR TITLE
Added unit tests for structures module

### PR DESCRIPTION
**Pull request by Stranger6667**

### CODE REVIEW

File: tests/test_structures.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    def dict_instance(self):
+        instance = CaseInsensitiveDict()
+        instance['Accept'] = 'application/json'
+        return instance
+
+    def test_list(self, dict_instance):
+        assert list(dict_instance) == ['Accept']
+
+    def test_getitem(self, dict_instance):
+        assert dict_instance['aCCept'] == 'application/json'
```
Comment (Lukasa) [inline tests/test_structures.py]: For this kind of test it's really valuable to have quite a few examples with wildly different cases. Something like:

``` python
@pytest.mark.parametrize('key', ['aCCept', 'Accept', 'accept', 'ACCEPT', 'aCcEpT'])
def test_getitem(self, dict_instance, key):
    assert dict_instance[key] == 'application/json'
```

```
```

File: tests/test_structures.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    def test_delitem(self, dict_instance):
+        del dict_instance['aCcept']
+        assert 'aCCept' not in dict_instance
+        assert 'Accept' not in dict_instance
```
Comment (Lukasa) [inline tests/test_structures.py]: Similarly, this can be parametrised.

**Comment by Lukasa** [comment]
@sigmavirus24 @kennethreitz Can one of you two do a separate code review of this? @Stranger6667 is here with me at PyCon SK writing some of these tests. I'm happy with these, but it'd be good if one of you two gave the ok/not-ok.

File: requests/structures.py [hunk 3]
```diff
@@ -85,6 +85,7 @@ def copy(self):
     def __repr__(self):
         return str(dict(self.items()))
 
+
```
Comment (sigmavirus24) [inline requests/structures.py]: Again, the changes in this file are not unit tests, they're strictly cosmetic

File: tests/test_structures.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
+# coding: utf-8
+import pytest
+
+from requests.structures import CaseInsensitiveDict, LookupDict
+
+
+class TestCaseInsensitiveDict:
+
+    @pytest.fixture
+    def dict_instance(self):
```
Comment (sigmavirus24) [inline tests/test_structures.py]: This doesn't use the `self` parameter and as such shouldn't be defined on the class. There should also be a docstring here explaining what dict_instance does for a quick glance.

In fact nothing in this class uses `self` and as such each should be its own function.

File: tests/test_structures.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            ({'AccePT': 'application/json'}, True),
+            ({}, False),
+            (None, False)
+        )
+    )
+    def test_equality(self, dict_instance, other, result):
+        assert (dict_instance == other) is result
+
+
+class TestLookupDict:
```
Comment (sigmavirus24) [inline tests/test_structures.py:53]: Same feedback about this class. Use more descriptive function names and make these all independent functions please.

Alternatively, don't use fixtures passed into each method, but instead have a "set-up" fixture that runs for each method and defines the `instance` on `self`.

That would justify fairly bland test names grouped inside classes.

**Comment by Stranger6667** [comment]
Hello @Lukasa @sigmavirus24 !
I've updated this PR :)

File: tests/test_structures.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+class TestCaseInsensitiveDict:
+
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        """
+        CaseInsensitiveDict instance with "Accept" header.
+        """
+        self.case_insensitive_dict = CaseInsensitiveDict()
+        self.case_insensitive_dict['Accept'] = 'application/json'
```
Comment (Lukasa) [inline tests/test_structures.py:15]: Ah, sorry @Stranger6667, but I think @sigmavirus24 wanted [something more like this](https://pytest.org/latest/xunit_setup.html#method-and-function-level-setup-teardown).

File: tests/test_structures.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+class TestLookupDict:
+
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        """
+        LookupDict instance with "bad_gateway" attribute.
+        """
+        self.lookup_dict = LookupDict('test')
+        self.lookup_dict.bad_gateway = 502
```
Comment (Lukasa) [inline tests/test_structures.py:61]: Same here.

**Comment by Lukasa** [comment]
Awesome, we're very close now @Stranger6667. One small note. =)

**Comment by kennethreitz** [comment]
Looks pretty solid!

**Pull request merged**